src/packages/sdk/src/utils/loadEnv.ts

```diff
@@ -116,33 +83,57 @@ function readSchemaPathFromPkgJson(): string | null {
   }
 }
 
-function tryLoadEnvFromPkgJson(
-  schemaPath: string,
-  cwd: string,
-): LoadEnvResult | null {
-  try {
-    const dotenvFilepath = path.join(path.dirname(schemaPath), '.env')
-
-    if (!fs.existsSync(schemaPath) || !fs.existsSync(dotenvFilepath)) {
-      debug(
-        'Environment variables not loaded (package.json configuration was provided)',
-      )
-      return null
+/**
+ * Will throw an error if the file at `envPath` has env conflicts with `rootEnv`
+ */
+function checkForConflicts(
+  rootEnvInfo: LoadEnvResult | null,
+  envPath: string | null,
+) {
+  const notTheSame = rootEnvInfo?.path && envPath && path.resolve(rootEnvInfo?.path) !== path.resolve(envPath)
+  const parsedRootEnv = rootEnvInfo?.dotenvResult.parsed
+  if (parsedRootEnv && envPath && notTheSame && fs.existsSync(envPath)) {
+    const envConfig = dotenv.parse(fs.readFileSync(envPath))
+    const conflicts: string[] = []
+    for (const k in envConfig) {
+      if (parsedRootEnv[k] === envConfig[k]) {
+        conflicts.push(k)
+      }
     }
+    if (conflicts.length > 0) {
+      throw new Error(`
+      You are trying to load env variables which are already present in your project root .env
+      \tRoot: ${rootEnvInfo?.path}
+      \tPrisma: ${envPath}
+      \tEnv Conflicts:
+      ${conflicts.map((conflict) => `\t\t${conflict}`).join('\n')}
+
+      We would suggest that you move all your env vars to your projects root .env and remove the .env in ${envPath}
```

Comment:
```suggestion
      You can fix this by removing the .env file from "${envPath}" and move its contents to your .env file at the root "${rootEnvInfo?.path}"
```

Maybe?

Reply:
but the `?` in rootEnvInfo? is a bit weird, that means it could be undefined?

Reply:
https://github.com/prisma/prisma/pull/3971/files/3ba400dd4e2a975c72412e4483bc5602da552fbe#diff-59890438ec39361cbc9c8bf8a8772c311453c15035dd4e82351b04d70980fcf4R94-R95